agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/post-combat-game-state/resolve-retreat-game-state/ResolveRetreatGameState.ts

```diff
@@ -47,9 +47,11 @@ export default class ResolveRetreatGameState extends GameState<
     }
 
     firstStart(): void {
-        const possibleRetreatRegions = this.getValidRetreatRegions();
+        const regionsWithTheirCasualties: BetterMap<Region, number> = new BetterMap<Region, number>(
```

Comment:
Could you modify `getValidRetreatRegions`with the code to compute the list of valid retreat regions, instead of putting the logic here?

Modifying this function would also make the UI highlight the correct retreat regions.

Reply:
> In fact the valid retreat regions are already shown by the UI but of course... I can modify it but it will take till afternoon

Indeed, you're right! I forgot that SelectRegion would take care of that.

You can still put the changes in the `getValidRetreatRegions` to make the code more clearer. Also, some comments to describe which part of the code is responsible for computing the lowest number of casualties and which part of the code is responsible for computing the final list of retreat regions.

Also, you don't need to use a `BetterMap` for this, you can directly use a list of tuples.

Otherwise, LGTM!

Reply:
In fact the valid retreat regions are already shown by the UI but of course... I can modify it but it will take till afternoon

Reply:
Done